catroid/src/main/java/org/catrobat/catroid/ui/controller/LookController.java

```diff
@@ -468,35 +471,64 @@ public void loadPictureFromLibraryIntoCatroid(String filePath, Activity activity
 		pictureOnSdCard.delete();
 	}
 
-	public boolean checkIfPocketPaintIsInstalled(Intent intent, final Activity activity) {
-		// Confirm if Pocket Paint is installed else start dialog --------------------------
-
-		List<ResolveInfo> packageList = activity.getPackageManager().queryIntentActivities(intent,
+	public boolean isPocketPaintInstalled(Activity activity, Intent intent) {
+		List<ResolveInfo> packages = activity.getPackageManager().queryIntentActivities(intent,
 				PackageManager.MATCH_DEFAULT_ONLY);
 
-		if (packageList.size() <= 0) {
-			AlertDialog.Builder builder = new CustomAlertDialogBuilder(activity);
-			builder.setTitle(R.string.pocket_paint_not_installed_title);
-			builder.setMessage(R.string.pocket_paint_not_installed).setCancelable(false)
-					.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
-						@Override
-						public void onClick(DialogInterface dialog, int id) {
-							Intent downloadPocketPaintIntent = new Intent(Intent.ACTION_VIEW, Uri
-									.parse(Constants.POCKET_PAINT_DOWNLOAD_LINK));
-							activity.startActivity(downloadPocketPaintIntent);
-						}
-					})
-					.setNegativeButton(R.string.no, new DialogInterface.OnClickListener() {
-						@Override
-						public void onClick(DialogInterface dialog, int id) {
-							dialog.cancel();
-						}
-					});
-			AlertDialog alert = builder.create();
-			alert.show();
-			return false;
-		}
-		return true;
+		return !packages.isEmpty();
+	}
+
+	public void installAndOpenPocketPaint(final Intent pocketPaintIntent, final Activity activity, final int
```

Comment:
I really hate having to pass through all these parameters, but the current structure (with the controller/fragment/adapter and dialogs) doesn't really allow for a more elegant way. The whole setup should (and hopefully will) be changed anyway, so I did not really see any reason to spend too much time on it.